daemon/daemon.go

```diff
@@ -802,10 +805,25 @@ func NewDaemon(c *Config) (*Daemon, error) {
 			// node-name automatically derived
 			nodeaddress.SetName(nodeName)
 
-			// Try to retrieve node's cidr from k8s's configuration
-			if err := k8s.UseNodeCIDR(d.k8sClient, nodeName); err != nil {
-				return nil, fmt.Errorf("unable to retrieve node CIDR: %s", err)
+			// Try to retrieve node's cidr and addresses from k8s's configuration
+			k8sNode, err = d.k8sClient.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
```

Comment:
It can't, `ParseNode()` is used on 2 other places where it only parses the node, without the needed to re-fetch it from the kubernetes api-server.